src/bin/pg_autoctl/debian.c

```diff
@@ -0,0 +1,699 @@
+										  PG_CONFIG_TYPE_DEBIAN))
+	{
+		/* so we're dealing with a "normal" Postgres installation */
+		memcpy(pgConfigFiles, &debianConfFiles, sizeof(PostgresConfigFiles));
+
+		return true;
+	}
+
+	/* well that's all we know how to detect at this point */
+	return false;
+}
+
+
+/*
+ * debian_init_postgres_config_files initialises the given PostgresConfigFiles
+ * structure with the location of existing files as found on-disk given a
+ * Postgres configuration kind.
+ */
+static bool
+debian_init_postgres_config_files(PostgresSetup *pgSetup,
+								  PostgresConfigFiles *pgConfigFiles,
+								  PostgresConfigurationKind confKind)
+{
+	const char *pgdata = pgSetup->pgdata;
+
+	switch (confKind)
+	{
+		case PG_CONFIG_TYPE_UNKNOWN:
+		{
+			/* that's a bug really */
+			log_error("BUG: debian_init_postgres_config_files "
+					  "called with UNKNOWN conf kind");
+			return false;
+		}
+
+		case PG_CONFIG_TYPE_POSTGRES:
+		{
+			initPostgresConfigFiles(pgdata, pgConfigFiles,
+									PG_CONFIG_TYPE_POSTGRES);
+
+			if (postgresConfigFilesAllExist(pgConfigFiles))
+			{
+				return true;
+			}
+			else
+			{
+				/*
+				 * WARN the user about the unexpected nature of our setup here,
+				 * even if we then move on to make it the way we expect it.
+				 */
+				log_warn("Failed to find Postgres configuration files in PGDATA, "
+						 "as expected: \"%s\" does not exists",
+						 pgConfigFiles->conf);
+				return false;
+			}
+		}
+
+		case PG_CONFIG_TYPE_DEBIAN:
+		{
+			DebianPathnames debPathnames = { 0 };
+
+			if (!buildDebianDataAndConfDirectoryNames(pgSetup, &debPathnames))
+			{
+				log_warn("Failed to match PGDATA at \"%s\" with a debian "
+						 "setup following the data_directory template "
+						 "'/var/lib/postgresql/%%v/%%c'",
+						 pgSetup->pgdata);
+				return false;
+			}
+
+			initPostgresConfigFiles(debPathnames.confDirectory,
+									pgConfigFiles, PG_CONFIG_TYPE_DEBIAN);
+
+			if (postgresConfigFilesAllExist(pgConfigFiles))
+			{
+				return true;
+			}
+			else
+			{
+				/*
+				 * WARN the user about the unexpected nature of our setup here,
+				 * even if we then move on to make it the way we expect it.
+				 */
+				log_warn("Failed to find Postgres configuration files in PGDATA, "
+						 "as expected: \"%s\" does not exists",
+						 pgConfigFiles->conf);
+				return false;
+			}
+		}
+	}
+}
+
+
+/*
+ * buildDebianDataAndConfDirectoryNames builds the debian specific directory
+ * pathnames from the pgSetup pgdata location.
+ *
+ * For a debian cluster, we first have to extract the "cluster" name (%c) and
+ * then find the configuration files in /etc/postgresql/%v/%c with %v being the
+ * version number.
+ *
+ * Note that debian's /etc/postgresql-common/createcluster.conf defaults to
+ * using the following setup, and that's the only one we support at this
+ * moment.
+ *
+ *   data_directory = '/var/lib/postgresql/%v/%c'
+ *
+ */
+static bool
+buildDebianDataAndConfDirectoryNames(PostgresSetup *pgSetup,
```

Comment:
we are just translating `/var/lib/postgresql/%VERSION%/%CLUSTER%` to `/etc/postgresql/%VERSION%/%CLUSTER%` 
Looks like this is too much effort.

Reply:
Any debian user can change `data_directory` in `/etc/postgresql/createcluster.conf` and use something else that those paths, or in another ordering, or etc. So I though later we might want to parse `createcluster.conf` and use whatever is setup there. Currently we only have their default hard-coded and that might be enough.